### DISCUSSION TITLE
[Issue] Video & Sound stuttering.

### DISCUSSION THREAD

**Discussion by Vargavintern** [KnuckleCracker/CW4-bug-tracker#1078] [General]
Hey. I just got Creeper World 4 last night and as soon as I started playing the tutorial i noticed that the game had video and audio stuttering ever 2-3 secconds. When I move the camera left and right over the field you can see the stuttering coming in waves. I tried different resolutions and fiddles with all the different visual effects, everything from max visuals to max performance and nothing changes or impacts the stuttering. Turning off the "high fps" setting makes the stuttering harder to notice since the fps is lower but thats that.

[DxDiag.txt](https://github.com/KnuckleCracker/CW4-bug-tracker/files/10334538/DxDiag.txt)

**Comment by knucracker** [comment]
If you have not already, do a full system reboot (not a hibernate and restore).  That fixes these issues about 9 out of 10 times.  
There's nothing about your system specs that should created a stutter.  People with lower performance machines play the game without performance issues.    I also see you are running a recent video driver.

If a total reboot doesn't resolve it, the next most likely candidates are interfering software or drivers.  Audio drivers from additional sound devices have show up in similar reports across other games over the years.  Capture software (screen recording) has also been the culprit from time to time.  You can bring up task manager and watch the performance of the tasks as you run the game to see if anything pops to the top  every few seconds.

**Comment by Vargavintern** [comment]
I rebootet, got the latest drivers and I killed off ever program in the task manager that I dont need. Still the same stuttering, I see no changes in the task manager that anything should strain the system or that are showing up as spikes in the readings.

**Comment by knucracker** [comment]
Other things you can try:
- Run the game either full screen or in a window (the opposite of what you have currently tested under). When running in a window, make the window larger or smaller and note the impact if any.
- Pause the game. Does the stutter happen while the game is paused?  (Check for audio or mouse movement stutter).
- Check your system for additional audio output or input devices you might not be aware of. Drivers or management software for audio devices have caused problems for other people over the years. If possible, temporarily disconnecting an external device to identify the source of the problem has helped. (Note that rebooting after disconnecting a device was sometimes necessary).
- If you have other games of a similar nature, try them and see if they have a stutter.

The game itself doesn't do anything special at intervals of a few seconds.  When running, it basically does mostly the same thing every game frame.  If something interferes with the audio, mouse movement, or freezes the screen every few seconds it has to be pretty low level on the system

**Reply by knucracker** [reply]
One other thing... if you are on Steam, disable the steam overlay for the game.  Right click the game in the steam library list of games and go to properties. Uncheck the "enable steam overlay...".  The overlay has caused GPU issues for some people in the past.

**Reply by Vargavintern** [reply]
I disabled the steam overlay and disconnected all my non-essential USB's and rebooted my computer. Tried window/fullscreen in different resolution, quit all applications and just ran with only steam and my firewall and still the same stuttering. The mouse cursor never stutters just the camera. The stuttering seems almost rytmic and comes in waves, the stuttering never cease but its bit random intervals that differs a second or so added between the waves.

**Comment by knucracker** [comment]
Take a look at the game's log file.  Maybe there is some hint in there.  Probably not since everything is running, just not smoothly.  But can't hurt to look.  The log file is here:
"c:\Users\YOUR_USER\AppData\LocalLow\Knuckle Cracker\Creeper World 4\Player.log"
It gets created from scratch each time you run the game.
